src/MessageBodyTypes.ts

```diff
@@ -105,6 +194,36 @@ export function parseBody(type: MessageBodyType, data: object): BodyType {
         case 'OneDeviceStatus':
             theType = OneDeviceStatus;
             break;
+        case 'OneZoneDefinition':
+            theType = OneZoneDefinition;
+            break;
+        case 'OneAreaDefinition':
+            theType = OneAreaDefinition;
+            break;
+        case 'OneAreaStatus':
+            theType = OneAreaStatus;
+            break;
+        case 'MultipleAreaStatus':
+            theType = MultipleAreaStatus;
+            break;
+        case 'OneControlStationDefinition':
+            theType = OneControlStationDefinition;
+            break;
+        case 'OneProjectDefinition':
+            theType = OneProjectDefinition;
+            break;
+        case 'OneLinkDefinition':
+            theType = OneLinkDefinition;
+            break;
+        case 'OneLinkNodeDefinition':
+            theType = OneLinkNodeDefinition;
+            break;
+        case 'MultipleLinkNodeDefinition':
+            theType = MultipleLinkNodeDefinition;
+            break;
+        case 'MultipleLinkDefinition':
+            theType = MultipleLinkDefinition;
+            break;
```

Comment:
man, this is a crazy amount of boilerplate. there's gotta be a better way, I just don't know what it is.

Reply:
@thenewwazoo I'm with you, personally speaking, I'd pass most of this through to clients - but if the goal is to have structured API around this... "this is the way"

Reply:
(This does have the benefit over time of having the protocol documented in code, and usage :) )

Reply:
Shower thoughts: Another way is we could have a spec plist/xml that gets generated into code...

Reply:
It'd still be hand-crafted, it would just be one degree removed from the source, which feels icky. I think the main source of indirection is going from a string to an object (without, y'know, `eval` 😅). Oh well.